### PR TITLE
Organize CI pipelines to make test result directly readable

### PR DESCRIPTION
**Pull request by petrkotas**
### What this PR does / why we need it:

Split makefile to eliminate redundant calls, introduce `make pr` to call before pushing.
Separate CI jobs to eliminate redundant actions to make failures unique.

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### CODE REVIEW

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -6,51 +6,71 @@ variables:
...
+    displayName: 🕵️Check for missing codegen
+
+- job: Golang_Code_Lint
+  dependsOn: Check_Code_Generation
+  pool: ARO-CI
+  steps:
+  - template: ./templates/template-checkout.yml
+  - script: |
+      set -xe
+      make lint-go
```
Comment (gvanderpotte) [inline .pipelines/ci.yml:37]: I think here we should have the [[ -z "$(git status -s)" ]] to check about the effects of linter

File: Makefile [hunk 2]
```diff
@@ -104,29 +104,34 @@ e2e.test:
 test-e2e: e2e.test
 	./e2e.test -test.timeout 180m -test.v -ginkgo.v
 
-test-go: generate
-	go build -tags containers_image_openpgp ./...
+test-go:
```
Comment (gvanderpotte) [inline Makefile:107]: are not we missing the dependency on generate here ?

**Review by gvanderpotte — CHANGES_REQUESTED** [review]
a couple of questions here

File: Makefile (hunk 2 above)
Comment (petrkotas) [reply Makefile:107]: Yes we are and we are not, I am still figuring out how to piece it best. I would like us to call generate mindfully not implicitly during `make test`. I am now waiting for the ci and will iterate.

File: .pipelines/ci.yml (hunk 1 above)
Comment (petrkotas) [reply .pipelines/ci.yml:37]: Ah damn, you are right. I completely forgot that some linters do in-place changes.

**Comment by mjudeikis** [comment]
@petrkotas how many separate CI jobs you are thinking? Challenge here is that each of these jobs are running in individual CI node with parallelization =1. So by adding more jobs we basically making all CI slower for all PRs and nightly jobs due to that each workers will have a lot of smaller jobs to run. So spiting to a lot of jobs gives us other time related cost to all this. So at which point it is not worth it anymore?

**Comment by gvanderpotte** [comment]
> @petrkotas how many separate CI jobs you are thinking? Challenge here is that each of these jobs are running in individual CI node with parallelization =1. So by adding more jobs we basically making all CI slower for all PRs and nightly jobs due to that each workers will have a lot of smaller jobs to run. So spiting to a lot of jobs gives us other time related cost to all this. So at which point it is not worth it anymore?

I think the split done as it is with this PR is good enough: gives you in a glimpse where it failed, while not generating too many jobs that would slow the overall CI (we might not want to split much more than that now)

File: .pipelines/ci.yml [hunk 3]
```diff
@@ -6,51 +6,71 @@ variables:
...
   - script: |
       set -xe
-      make test-python
+      make generate
       [[ -z "$(git status -s)" ]]
-    displayName: 🧪Run Python Unit Tests
+    displayName: 🕵️Check for missing codegen
+
+- job: Golang_Code_Lint
+  dependsOn: Check_Code_Generation
```
Comment (mjudeikis) [inline .pipelines/ci.yml:31]: How this would work if scheduled on separate workers?

File: .pipelines/ci.yml [hunk 4]
```diff
@@ -6,51 +6,71 @@ variables:
...
+  pool: ARO-CI
+  steps:
+  - template: ./templates/template-checkout.yml
+  - script: |
+      set -xe
+      make lint-go
+    displayName: 🕵️Run Golang Linters
 
 - job: Golang_Unit_Tests
+  dependsOn: Check_Code_Generation
```
Comment (mjudeikis) [inline .pipelines/ci.yml:41]: Same as above

**Comment by mjudeikis** [comment]
@petrkotas considering separate workers and parallelization challenges do you want to progress with this one?

**Comment by petrkotas** [comment]
Closing in favor of #1465

**Pull request closed without merging**